src/features/Folding.ts

```diff
@@ -421,9 +421,9 @@ export class FoldingProvider implements vscode.FoldingRangeProvider {
     ): ITokenList {
         const result = [];
 
-        const emptyLine = /^[\s]+$/;
-        const startRegionText = /^#\s*region\b/;
-        const endRegionText = /^#\s*endregion\b/;
+        const emptyLine = /^[\s]*$/;
+        const startRegionText = /^#\s*[rR]egion\b/;
```

Comment:
Does JS support `(?i)` to set the regex to ignore case?

Reply:
Depends.  This commit brings it into line with the VSCode "default" behaviour, but, we could add this.  I'm not fussed either way on this issue.

Reply:
Good enough for me. FWIW, ISE only supports lower-case `#region / #endregion`.

Reply:
OTOH leading folks down a path that doesn't work in ISE … is that a bad thing or if so, do we care?  Also, ISE doesn't like a space between the `#` and `region|endregion`.  It's pretty picky.

Reply:
Agreed!

Reply:
Mostly agreed.  I've updated the regexes however I've kept the "space between `#` and `region`" e.g. `# region` is allowed.

While ISE and other langauges don't allow a space, the existing VS Code default for powershell is to allow zero or more space (https://github.com/Microsoft/vscode/blob/master/extensions/powershell/language-configuration.json#L26-L31).  Removing that is a kinda-breaking change.

Reply:
Not sure I agree with the VSCode defaults.  Adding the space allows me to write a normal comment that just happens to begin with the word region but isn't the start of a "region" e.g. 
```
# blah blah blah blah in the same 
# region as blah blah blah
```

Reply:
yeah .... could also argue the same for;

```
#blah blah blah blah in the same 
#region as blah blah blah
```

Reply:
Except that if I type a comment like `#Requires that the user run as administrator` at the top of my file, I can fix it by adding the space e.g. `# Requires that the user run as administrator`

Reply:
True....

Reply:
Interesting the tests for that, don't include spaces between `#`
https://github.com/Microsoft/vscode/commit/9e05d4b635b403cdb94bb5fe6600e1d9710ab320#diff-e10026493ca8c706f842f8b49b1ccdfa

Reply:
I also find it odd that they allow any number of spaces after the `#` e.g. `#                Region` would be a valid region start.

Reply:
So given all this, plus the powershell region snippet has no space (https://github.com/PowerShell/vscode-powershell/blob/705f1747844a8f11f187cd008622f175094c2b72/snippets/PowerShell.json#L947-L955) I'll change the regex to not allow whitespace.

Can always add it back in :-)

Reply:
Fixed.